react/react1/week3/week3/src/component/TodoItem.jsx

```diff
@@ -0,0 +1,52 @@
+/** @format */
+
+import React, { useState } from 'react';
+
+function TodoItem({ todo, onchange, onDone, onDelete }) {
+  const [isEditing, setIsEditing] = useState(false);
+
+  let content;
+  if (isEditing) {
+    content = (
+      <div>
+        <input
+          value={todo.description}
+          onChange={(e) => {
+            onchange({
+              ...todo,
+              description: e.target.value
+            });
+          }}
+        />
+      
+        <button onClick={() => setIsEditing(false)}> update</button>
+      </div>
+    );
+  } else {
+    content = (
+      <div>
+        {todo.description} | {todo.deadline}
+        <button onClick={() => setIsEditing(true)}> Edit</button>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <li
+        className="todo-list"
+        key={todo.id}>
+        <input
+          type="checkbox"
+          checked={todo.done}
+          onChange={() => onDone(todo.id)}
+        />
+        <p style={todo.done ? { textDecoration: 'line-through' } : {}}>
+          {content}
```

Comment:
Good!

Reply:
Thanks, Lucas for the review. I will update the fetch as you recommended.